lib/WUI/nhttp/file_info.cpp

```diff
@@ -82,7 +149,45 @@ JsonResult FileInfo::DirRenderer::renderState(size_t resume_point, JsonOutput &o
 }
 
 JsonResult FileInfo::FileRenderer::renderState(size_t resume_point, JsonOutput &output, FileInfo::FileState &state) const {
-    char *filename = state.owner->filename;
+    if (api == APIVersion::Octoprint)
+        return renderStateOctoprint(resume_point, output, state);
+    else
+        return renderStateV1(resume_point, output, state);
+}
+
+JsonResult FileInfo::FileRenderer::renderStateV1(size_t resume_point, JsonOutput &output, FileInfo::FileState &state) const {
+    char *filename = state.owner->filepath;
+    get_SFN_path(filename);
+    JSONIFY_STR(filename);
+    char long_name[FILE_NAME_BUFFER_LEN];
+    get_LFN(long_name, sizeof long_name, filename);
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+    JSON_OBJ_START;
+
+        JSON_FIELD_STR("name", basename_b(filename_escaped)) JSON_COMMA;
+        JSON_FIELD_BOOL("ro", false) JSON_COMMA;
+        JSON_FIELD_STR("type", file_type_by_ext(long_name)) JSON_COMMA;
+        JSON_FIELD_INT("m_timestamp", state.m_timestamp) JSON_COMMA;
+        JSON_FIELD_INT("size", state.size) JSON_COMMA;
+        JSON_FIELD_STR("display_name", long_name) JSON_COMMA;
+        if (filename_is_gcode(filename_escaped)) {
```

Comment:
Again, we probably should use the FS functions on the original, not on a JSON-escaped version.